deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJobIT.java

```diff
@@ -476,6 +476,50 @@ public void resumeCompletedWorkIngestTest() throws Exception {
         ingestedObjectsCount(3);
     }
 
+    @Test
+    public void ingestWorkObjectObjCountWithRetry() throws Exception {
+        String label = "testwork";
+        PID workPid = pidMinter.mintContentPid();
+
+        // Construct the deposit model with work object
+        Model model = job.getWritableModel();
+        Bag depBag = model.createBag(depositPid.getRepositoryPath());
+
+        // Constructing the work in the deposit model with a label
+        Bag workBag = model.createBag(workPid.getRepositoryPath());
+        workBag.addProperty(RDF.type, Cdr.Work);
+        workBag.addProperty(CdrDeposit.label, label);
+
+        PID mainPid = addFileObject(workBag, FILE1_LOC, FILE1_MIMETYPE, FILE1_SHA1, FILE1_MD5);
+        PID mainPid2 = addFileObject(workBag, FILE1_LOC, FILE1_MIMETYPE, null, FILE1_MD5_BAD);
+
+        depBag.add(workBag);
+
+        workBag.asResource().addProperty(Cdr.primaryObject,
+                model.getResource(mainPid.getRepositoryPath()));
+
+        workBag.asResource().addProperty(Cdr.primaryObject,
+                model.getResource(mainPid2.getRepositoryPath()));
+
+        job.closeModel();
+
+        try {
+            job.run();
+            fail("Test should not reach this line. Job should have thrown exception");
+        } catch (JobFailedException e) {
+            Map<String, String> jobStatus = jobStatusFactory.get(jobUUID);
+            assertNull(jobStatus.get(JobField.num.name()));
+            Resource fileResc =job.getWritableModel().getResource(mainPid2.getRepositoryPath());
+            fileResc.addProperty(CdrDeposit.md5sum, FILE1_MD5);
```

Comment:
You also need to remove the bad md5sum property, otherwise the test might flap since there will be two md5sums and it might randomly pick the bad one.